### PR TITLE
[main][Feat] upgrade delta vault03 deploy script

### PR DESCRIPTION
**Pull request by sundayalpaca**
## Description

- DeltaVault deployment script
- DeltaVault upgrade script BSC and FTM

### CODE REVIEW

File: .openzeppelin/unknown-250.json [hunk 1]
```diff
@@ -5757,13 +5757,13 @@
           {
             "contract": "DeltaNeutralVault",
             "label": "priceOracle",
-            "type": "t_contract(IDeltaNeutralOracle)10651",
```
Comment (alpanaca) [inline .openzeppelin/unknown-250.json:5760]: Don't think we need this file

File: package.json [hunk 2]
```diff
@@ -313,6 +316,7 @@
     "deploy:fantom_mainnet:revenue-treasury:deploy:revenue-treasury": "hardhat --network fantom_mainnet deploy --no-compile --reset --tags RevenueTreasury",
     "deploy:mainnetfork": "hardhat --network mainnetfork deploy --no-compile --reset",
     "deploy:fantom_mainnetfork": "hardhat --network fantom_mainnetfork deploy --no-compile --reset",
+    "deploy:fantom_mainnetfork:delta-neutral-vault:upgrade:delta-neutral-vault": "hardhat --network fantom_mainnetfork deploy --no-compile --reset --tags UpgradeDeltaNeutralVaultFTM",
```
Comment (alpanaca) [inline package.json:319]: don't need this

File: deploy/exec/delta-neutral-vault-config/deploy/delta-neutral-vault-config02.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   let config = configFileHelper.getConfig();
 
   // use to write config file
-  const DELTA_VAULT_SYMBOL = "L8x-USDTBNB-PCS1";
+  const DELTA_VAULT_SYMBOL = "n8x-BNBUSDT-PCS3";
```
Comment (alpanaca) [inline deploy/exec/delta-neutral-vault-config/deploy/delta-neutral-vault-config02.ts:27]: Don't need to commit this

File: deploy/exec/delta-neutral-vault/deploy/delta-neutral-vault03.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  ];
+
+  const alpacaTokenAddress = config.Tokens.ALPACA;
+  const wNativeRelayerAddr = config.SharedConfig.WNativeRelayer;
+  for (let i = 0; i < deltaVaultInputs.length; i++) {
+    const deltaVaultInput = deltaVaultInputs[i];
+    const stableVault = config.Vaults.find((v) => v.symbol === deltaVaultInput.stableVaultSymbol);
+    const assetVault = config.Vaults.find((v) => v.symbol === deltaVaultInput.assetVaultSymbol);
+    if (stableVault === undefined) {
+      throw `error: unable to find vault from ${deltaVaultInput.stableVaultSymbol}`;
```
Comment (almost-llamas) [inline deploy/exec/delta-neutral-vault/deploy/delta-neutral-vault03.ts:62]: Do not throw string, please throw an Error object.

File: deploy/exec/delta-neutral-vault/deploy/delta-neutral-vault03.ts [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    }
+    if (assetVault === undefined) {
+      throw `error: unable to find vault from ${deltaVaultInput.assetVaultSymbol}`;
+    }
+
+    if (!deltaVaultInput.deltaNeutralVaultConfig) {
+      const deltaVault = config.DeltaNeutralVaults.find((dv) => dv.symbol === deltaVaultInput.symbol);
+      if (!deltaVault) throw Error(`Couldn't find DeltaNeutralVaults[${deltaVaultInput.symbol}]`);
+      if (!validateAddress(deltaVault.config))
+        throw Error(`DeltaNeutralVaults[${deltaVaultInput.symbol}] > config (${deltaVault.config}) address is invalid`);
```
Comment (almost-llamas) [inline deploy/exec/delta-neutral-vault/deploy/delta-neutral-vault03.ts:72]:
```
throw new Error('..')

File: deploy/exec/delta-neutral-vault/upgrade/delta-neutral-vault-ftm.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+  const converter = new Converter();
+  const toBeUpgradedVaults = converter.convertDeltaSymboltoObj(TARGETED_VAULTS);
+  let nonce = await deployer.getTransactionCount();
+
+  for (const vault of toBeUpgradedVaults) {
+    console.log("------------------");
+    console.log(`> Upgrading DeltaNeutralVault at ${vault.symbol} through Timelock + ProxyAdmin`);
+    console.log("> Prepare upgrade & deploy if needed a new IMPL automatically.");
+    const NewVault = await ethers.getContractFactory(DELTA_NEUTRAL_VAULT);
```
Comment (almost-llamas) [inline deploy/exec/delta-neutral-vault/upgrade/delta-neutral-vault-ftm.ts:45]: newVault

**Review by almost-llamas — APPROVED** [review]

**Review by jakeal-papaca — APPROVED** [review]

**Pull request merged**